pylint/checkers/base.py

```diff
@@ -139,6 +139,19 @@ def _is_multi_naming_match(match, node_type, confidence):
             match.lastgroup not in EXEMPT_NAME_CATEGORIES
             and (node_type != 'method' or confidence != interfaces.INFERENCE_FAILURE))
 
+def is_typing_type(obj):
```

Comment:
Maybe this should be a brain plugin rather than a specific inference function here? We just need to infer these as classes.